app/controllers/admin/bulk_uploads_controller.rb

```diff
@@ -26,7 +26,7 @@ def create
     @bulk_upload.attachments_attributes = create_params[:attachments_attributes]
     @bulk_upload.attachments.each do |attachment|
       attachment.attachment_data.attachable = @edition
-      attachment.attachment_data.use_non_legacy_endpoints = use_non_legacy_endpoints?
+      attachment.attachment_data.use_non_legacy_endpoints = use_non_legacy_endpoints? unless attachment.attachment_data.csv?
```

Comment:
it would be a bit cleaner if the decision about csvs would be in the `use_non_legacy_endpoints?` method. it can take attachment_data as an argument.